src/PSRule.Rules.Azure/rules/Azure.Redis.Rule.ps1

```diff
@@ -30,6 +30,60 @@ Rule 'Azure.Redis.MaxMemoryReserved' -Type 'Microsoft.Cache/Redis' -Tag @{ relea
     $Assert.GreaterOrEqual($TargetObject, 'Properties.redisConfiguration.maxmemory-reserved', $memSize * 0.1, $True);
 }
 
+# Synopsis: Premium and Enterprise Redis cache should be deployed with availability zones for high availability.
+Rule 'Azure.Redis.AvailabilityZone' -Type 'Microsoft.Cache/Redis' -If { (IsPremiumCache) -or (IsEnterpriseCache) } -Tag @{ release = 'GA'; ruleSet = '2021_12'; } {
```

Comment:
Should this be one or two rules? If one we should probably add an additional type here `Microsoft.Cache/redisEnterprise`.

Reply:
@BernieWhite Good catch. I will include this type as well. I didn't create a `Microsoft.Cache/redisEnterprise` during my testing so missed this type. 

We could create another rule for this, would make things simpler. What should we call it? `Azure.Redis.EnterpriseAvailabilityZone` 😄

Reply:
We could even split by SKU like `Azure.Redis.PremiumAvailabilityZone` and `Azure.Redis.EnterpriseAvailabilityZone`.

Reply:
Might be worth spliting into `Azure.EnterpriseRedis` since other rules will need to differentiate down the line.

Reply:
Added changes for this, let me know what you think.

Reply:
@ArmaanMcleod That makes sense to me. Is treated as a different resource.